src/index.js

```diff
@@ -11,16 +12,28 @@ import parseUrl from 'url-parse';
   * This document aims to describe all the functions supported by the SDK, as well as showing examples of their expected usage.
   *
   * If you feel something is missing, not clear or could be improved, please don't hesitate to open an [issue in GitHub](https://github.com/agile-iot/agile-sdk/issues/new), we'll be happy to help.
-  * @param {string} - agile-core REST API endpoint
+  * @param {Object} - including the api base url, idm base url (if not provided defaults to the same api host but default idm port), and a token.
   * @returns {Object}
   * @example
-  * var agile = require('agile-sdk')('http://agile-core:8080')
+  * var agile = require('agile-sdk')({
+      api:'http://agile-core:8080',
+      idm: 'http://agile-core:3000',
+      token: 'zIOycOqbEQh4ayw7lGAm9ILBIr'
+    })
 */
-const agileSDK = (base) => {
+const agileSDK = (params) => {
+  //backwards compatibility
+  if (typeof params === 'string' || params instanceof String){
+    params = {api: params};
```

Comment:
I don't understand your point. What is not working as expected?
I have tried to execute the agile-ui with this branch of the SDK and it runs.  It may be easier for you to just push an update. If so, please feel free to do it.

I have also updated the documentation again to fix an inconsistency with the sensor type (as you pointed out).